src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -28,7 +28,9 @@ public interface IUntypedObject
         /// </summary>
         IUntypedObject this[int index] { get; }
 
-        bool TryGetProperty(string value, out IUntypedObject result);
+        bool TryGetProperty(string value, out IUntypedObject result);
+
+        bool TrySetProperty(string name, IUntypedObject value);
```

Comment:
>IUntypedObject value [](http://example.com/codeflow?start=41&length=20)

This could be a FormulaValue

Ie, it's ok to Set(uo.field, 123) 
where 123 is a known strongly typed value.